chainer_compiler/elichika/onnx_converters.py

```diff
@@ -317,7 +317,9 @@ def convert_node_bin_op(onnx_graph, node: 'nodes.NodeBinOp'):
         seq_right = right.create_sequence()
         onnx_graph.add_node(binops[node.binop], [seq_left, seq_right], [
                             value2onnx_parameter[node.outputs[0]].onnx_name], None)
-
+    elif isinstance(node.left, values.StrValue):
+        # Constant propagation, string concatenation, etc.
+        pass
```

Comment:
Cannot we raise `NotImplementedError` here? Silently ignoring expressions would be confusing. If we can change `pass` to `raise` here, I think it'd be better to change the above comment to a TODO comment.

Or, this `pass` is actually safe since we'll raise exception later in veval_bin_op? If it's the case, could you update the above comment to explain why it's OK to ignore binops for strings?

Reply:
veval_bin_op executes before onnx_converters. Since binary operations on strings types are being done in veval_bin_op itself (and incompatible types are raising assertion errors there), the generated NodeBinOp node doesn't have to do anything and hence is no-op. No need to raise any `NotImplementedError`.

Reply:
Yes, it should pass.

Reply:
It is passing already without emitting any onnx node, actually. Because elichika does the constant propagation while parsing.

Reply:
Hmm? I got the following error with this PR and my modification above
```
$ ./build/tools/run_onnx out/elichika_syntax_Print
Initializing ChainerX...
Loading model...
Loading data...
Found 1 test cases
Constructing model...
Failed to schedule (): ChainerPrint(@C_test_1, _L.8) -> ()
 _L.8 cannot be ready
Failed graph is stored in /tmp/chainer_compiler_failure.onnx
Check `false' failed! in CheckSanity at ../compiler/scheduler.cc:233:
zsh: abort (core dumped)  ./build/tools/run_onnx out/elichika_syntax_Print
```

Reply:
That is strange. For me, the test is passing just fine.

```
test 10
foo test 3
*** Testing out/elichika_tmp/tmp ***
build/tools/run_onnx --test out/elichika_tmp/tmp
Initializing ChainerX...
Loading model...
Loading data...
Found 1 test cases
Constructing model...
Generate code...
Running for out/elichika_tmp/tmp/test_data_set_0
Verifying the result...
OK: #returned_value
Elapsed: 0.285 msec
OK!
```

Reply:
This also failed on CI? https://github.com/pfnet-research/chainer-compiler/pull/481/commits/6ba836bdbde824309a12aa0f723a4deece082424

Reply:
Sorry, it had a different change. Updated link: https://github.com/pfnet-research/chainer-compiler/pull/481/commits/2ba3dc7810380d25ec461182885ef343551d8582

https://travis-ci.org/pfnet-research/chainer-compiler/builds/557706471?utm_source=github_status&utm_medium=notification

Reply:
I found the bug. Will submit a PR for the fix.